src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -127,6 +133,15 @@ private static void OnRequestSendRequestComplete(WinHttpRequestState state)
             state.TcsSendRequest.TrySetResult(true);
         }
 
+        private static void OnRequestDataAvailable(WinHttpRequestState state, int bytesAvailable)
+        {
+            Debug.Assert(state != null, "OnRequestDataAvailable: state is null");
```

Comment:
it could return 0.  That means 'end-of-file' and so, we do want to complete the task and then do the real `WinHttpReadData` call.

More info from the docs:

>This function returns the number of bytes of data that are available to read immediately by a subsequent call to WinHttpReadData. If no data is available and the end of the file has not been reached, then, the function returns TRUE, and when data becomes available, calls the callback function with WINHTTP_STATUS_CALLBACK_DATA_AVAILABLE and indicates the number of bytes immediately available to read by calling WinHttpReadData.